openam-authentication/openam-auth-oauth2/src/main/java/org/forgerock/openam/authentication/modules/oauth2/OAuthConf.java

```diff
@@ -267,6 +270,29 @@ public String getTokenServiceUrl(){
         return tokenServiceUrl;
     }
 
+    public Map<String, String> getTokenServicePOSTparametersForBasic(String code, String authServiceURL)
```

Comment:
It's not good to add similar code for maintainability.
For exaample, I think the exception message in getTokenServicePOSTparameters() is a bit wrong. To fix this, getTokenServicePOSTparametersForBasic() must be fixed too.

Since OAuthConf knows the client authentication method, it can select parameters in getTokenServicePOSTparameters().